var/spack/repos/builtin/packages/adept-utils/package.py

```diff
@@ -16,5 +17,10 @@ class AdeptUtils(CMakePackage):
     version('1.0',   sha256='fed29366c9bcf5f3799220ae3b351d2cb338e2aa42133d61584ea650aa8d6ff7')
 
     depends_on('boost@:1.72.0')
+
+    # TODO: replace this with an explicit list of components of Boost,
+    # for instance depends_on('boost +filesystem')
+    # See https://github.com/spack/spack/pull/22303 for reference
+    depends_on(Boost.with_default_variants)
```

Comment:
header-only

Reply:
header only can also just be a build dep, which is nice (:

Reply:
don't do this right now -- we really need a special `include` deptype.  The reason is that build dependencies are currently not included in the `dag_hash()`, so you'll get a bunch of ABI issues if you do not also make it a `link` dep.  The right deptype for boost right now is probably `link`, but like I said, we need something that will indicate that the dependency may affect the ABI.

Reply:
This will become more relevant very soon once we start more aggressively reusing installed packages -- we'll want the boost provenance so that we do not accidentally get things built with two boosts.

Reply:
Also, without `link` deptype, Spack won't automatically add `-I` via the compiler wrappers.